site/public/js/drag-and-drop.js

```diff
@@ -1145,12 +1131,11 @@ function handleUploadCourseMaterials(csrf_token, expand_zip, hide_from_students,
             try {
                 var jsondata = JSON.parse(data);
 
-                if (jsondata['status'] === 'success') {
-                    window.location.href = return_url;
-                }
-                else {
+                if (jsondata['status'] !== 'success') {
                     alert(jsondata['message']);
                 }
+                
+                window.location.href = return_url;
```

Comment:
why were these all changed to always redirect on failure? There's no mention in the PR as to why you're making changes to this file.

Reply:
Previously an error would be shown through and alert and the server would also display an error by $this->core>addError(); function but that would only display on page change.

I changed it to redirect which most cases would just be a page refresh to the same page so show the error once from the server, this would also prevent showing duplicate errors when the user moves changes to another page.

Reply:
We should try to not refresh / redirect under normal usage (like hitting max POST size). Stuff like CSRF token violation or uploading a file for someone that is not you is fine to refresh on since those are abnormal errors that should not really happen under normal usage, and if they do, at least in the case of the first error, a refresh is the only way to fix it.